js/app/interfaces/card.js

```diff
@@ -365,18 +365,25 @@ const Card = new Lang.Interface({
      *
      * Sets up a frame to show the model's thumbnail uri.
      */
-    set_thumbnail_frame_from_model: function (frame) {
+    set_thumbnail_frame_from_model: function (frame, background_size='cover') {
         frame.visible = false;
         if (!this.model.thumbnail_uri)
             return;
+        if (background_size === 'center') {
+            let im = new Gtk.Image({
```

Comment:
You sure about this? If a set or article has a really big image it would not be able to size down here. Seems like we could be triggering resolution support bugs (to big for composite) with content updates, which sounds scary to me.

Also nit: s/im/image/